src/Stratis.Features.FederatedPeg/Wallet/FederationWalletSyncManager.cs

```diff
@@ -208,6 +223,36 @@ public virtual void ProcessBlock(Block block)
             this.walletManager.ProcessBlock(block, newTip);
         }
 
+        /// <inheritdoc />
+        public virtual void ProcessBlock(Block block)
+        {
+            Guard.NotNull(block, nameof(block));
+
+            // If the queue reaches the maximum limit, ignore incoming blocks until the queue is empty.
+            if (!this.maxQueueSizeReached)
```

Comment:
Can we really simply ignore blocks?

Reply:
This is only going to happen in IBD, and what will happen is after the next batch of blocks comes in, the `OnProcessBlockAsync` method will retrieve any that are missing beforehand and continue where it left off.